packages/commonwealth/server/routes/deleteChain.ts

```diff
@@ -12,161 +13,248 @@ export const Errors = {
   NeedChainId: 'Must provide chain id',
   NoChain: 'Chain not found',
   CannotDeleteChain: 'Cannot delete a chain with registered addresses',
-  NotAcceptableAdmin: 'Not an Acceptable Admin'
+  NotAcceptableAdmin: 'Not an Acceptable Admin',
+  BadSecret: 'Must provide correct secret',
 };
 
-const deleteChain = async (models: DB, req: Request, res: Response, next: NextFunction) => {
-  if (!req.user) {
-    return next(new AppError(Errors.NotLoggedIn));
+type deleteChainReq = {
+  id: string;
+  airplaneSecret?: string;
+  airplaneManualSecret?: string;
+};
+
+type deleteChainResp = { result: string };
+
+const deleteChain = async (
+  models: DB,
+  req: TypedRequestBody<deleteChainReq>,
+  res: TypedResponse<deleteChainResp>,
+  next: NextFunction
+) => {
+  const { id, airplaneSecret, airplaneManualSecret } = req.body;
+
```

Comment:
we should add a list of chains here, before we merge, that can _never be deleted_. @jnaviask @dillchen @zakhap to help come up with a list